MainWindow.xaml.cs

```diff
@@ -30,15 +24,25 @@ public MainWindow()
 
             txtCount.Text = count.ToString();
         }
-
-        private void BtnTestDatabaseConn_OnClick(object sender, RoutedEventArgs e)
+        private void Window_OnLoaded(object sender, RoutedEventArgs e)
         {
-            DbConfigDialog dialog = new DbConfigDialog();
-            dialog.Owner = this;
+            if (!ValidateConfigXml(out string connStr))
```

Comment:
Here it should be put into a separate class and made public, otherwise the whole app startup is untestable.